File: gitdot-api/src/endpoint/user/list_user_organizations.rs
use serde::{Deserialize, Serialize};

use crate::{
    endpoint::Endpoint,
    resource::{OrganizationMemberResource, common::Page},
};

pub struct ListUserOrganizations;

impl Endpoint for ListUserOrganizations {
    const PATH: &'static str = "/user/{user_name}/organizations";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListUserOrganizationsRequest;
    type Response = ListUserOrganizationsResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListUserOrganizationsRequest {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub cursor: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub limit: Option<u32>,
}

pub type ListUserOrganizationsResponse = Page<OrganizationMemberResource>;

File: gitdot-web/packages/gitdot-api/src/endpoint/user/list-user-organizations.ts
import { z } from "zod";
import { page, UserOrganizationResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListUserOrganizationsRequest = z.object({
  cursor: z.string().optional(),
  limit: z.number().int().positive().optional(),
});
export type ListUserOrganizationsRequest = z.infer<
  typeof ListUserOrganizationsRequest
>;

export const ListUserOrganizationsResponse = page(UserOrganizationResource);
export type ListUserOrganizationsResponse = z.infer<
  typeof ListUserOrganizationsResponse
>;

export const ListUserOrganizations = {
  path: "/user/{user_name}/organizations",
  method: "GET",
  request: ListUserOrganizationsRequest,
  response: ListUserOrganizationsResponse,
} as const satisfies Endpoint;
export type ListUserOrganizations = typeof ListUserOrganizations;
